Core/Core/Serialisation/DeserializationWorkerThreads.cs

```diff
@@ -57,6 +56,11 @@ protected override void ThreadMain()
       catch (Exception ex)
       {
         tcs.SetException(ex);
+
```

Comment:
You might want to share this pattern more widely as I think it came up yesterday and this look like a better solution.

Reply:
Not convinced this pattern is needed outside of manually manipulating threads (which we should avoid doing, even inside core imo)

But if you have an example of where else this could be applied, happy to re-consider. You mentioned @clairekuang came a across some place that sounded similar?